.github/workflows/external-application-connection-tests.yml

```diff
@@ -0,0 +1,203 @@
+              -f /usr/share/pki/server/examples/installation/ca.cfg \
+              -s CA \
+              -D pki_cert_id_generator=random \
+              -D pki_request_id_generator=random \
+              -v
+
+      - name: Run PKI healthcheck
+        run: docker exec pki pki-healthcheck --debug
+
+      - name: Initialize PKI client
+        run: |
+          docker exec pki pki-server cert-export ca_signing --cert-file ca_signing.crt
+          docker exec pki pki client-cert-import ca_signing --ca-cert ca_signing.crt
+          docker exec pki pki info
+
+      - name: Check CA admin
+        run: |
+          docker exec pki pki client-cert-import \
+              --pkcs12 /root/.dogtag/pki-tomcat/ca_admin_cert.p12 \
+              --pkcs12-password Secret.123
+          docker exec pki pki -n caadmin ca-user-show caadmin
+
+      - name: Create postgresql certificates
+        run: |
+          docker exec pki pki nss-cert-request \
+              --subject "CN=postgresql.example.com" \
+              --ext /usr/share/pki/server/certs/sslserver.conf \
+              --csr sslserver.csr
+          REC_ID=$(docker exec pki pki ca-cert-request-submit  \
+              --profile caServerCert \
+              --csr-file sslserver.csr \
+              --subject "CN=postgresql.example.com" | grep "Request ID")
+          CERT_ID=$(docker exec pki pki -n caadmin ca-cert-request-approve ${REC_ID:14} --force | \
+              grep "Certificate ID")
+          docker exec pki pki ca-cert-export ${CERT_ID:18} --output-file sslserver.crt
+          docker exec pki pki nss-cert-import --cert sslserver.crt postgresql
+          docker exec pki pk12util -o sslserver.p12 -n postgresql -d /root/.dogtag/nssdb -W secret
+          docker cp pki:sslserver.p12 .
+          openssl pkcs12 -in sslserver.p12 -nocerts -out sslserver.key -noenc -password  pass:secret
+          openssl pkcs12 -in sslserver.p12 -nokeys -clcerts -out sslserver.crt  -password pass:secret
+          openssl pkcs12 -in sslserver.p12 -nokeys -cacerts -out ca.crt  -password pass:secret
+
+      - name: Build postgrsql image with certificates
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          tags: postgres-ssl
+          target: postgres-ssl
+
+      - name: Deploy postgresql
+        run: |
+          docker run -d --name postgresql \
+              --hostname postgresql.exampla.com \
+              -e POSTGRES_PASSWORD=mysecretpassword \
+              -e POSTGRES_USER=acme \
+              postgres-ssl -c ssl=on \
+              -c ssl_cert_file=/var/lib/postgresql/server.crt \
+              -c ssl_key_file=/var/lib/postgresql/server.key
+
+      - name: Connect DB container to network
+        run: docker network connect example postgresql --alias postgresql.example.com
+
+      - name: Set up database drivers
+        run: |
+          docker exec pki dnf install -y postgresql-jdbc ongres-scram-client ongres-stringprep
```

Comment:
To my understanding `ongres-scram-client` and `ongres-stringprep` will be installed automatically when we install `postgresql-jdbc`, so they don't need to be installed explicitly.